docs/fud/axi-gen.md

```diff
@@ -0,0 +1,72 @@
+# Axi generation
+
+Calyx currently generates a fairly complex AXI interface that can be daunting
+to deal with if confronting for the first time.
+The following is an overview of how the generation occurs
+and how the output AXI interface behaves as of 2022-9-11.
+
+In general, when `fud` is asked to create an [`.xclbin` file][xclbin] a `kernel.xml`,
+`main.sv`, and `toplevel.v` are created as intermediate steps for [xilinx tools][xilinx_tools]
+to properly work.
+
+`main.sv` contains the SystemVerilog needed for our computations to perform
+correctly. It is implemented as an FSM that derives from the original Calyx program.
+These files are very hard to read and not really conducive to being examined directly.
+[`kernel.xml`][kernel_xml] defines register maps and ports of our
+toplevel xilinx tools needs. `toplevel.v` wraps our computation kernel and contains
+the AXI interface for each memory define in a calyx program. 
+
+## Toplevel
+
+Our toplevel is generated through files in `src/backend/xilinx/`.
+
+### AXI memory controller
+Here, a calyx program is queried and all of its sub components are extracted.
```

Comment:
This is slightly confusing since only the toplevel component (link to https://docs.calyxir.org/lang/attributes.html?highlight=toplevel#toplevel) queried and only memories marked `@external` (link to https://docs.calyxir.org/lang/attributes.html?highlight=toplevel#external) are turned into AXI buses